# COMM318 Final Project

### Your final project should have three components:

#### 1. Your data

* Organize the various data files you have used in your analysis in the `data` folder.


* You can use sub-folders if this helps with the organization. 
    * For example, you might have:
        * an `orig` or `raw` folder for original data your downloaded from `data.gov`
        * a `clean` or `processed` folder that contains the cleaned up, selected and/or merged datasheets you created during your data analysis.


#### 2. Analysis notebooks

* A series of Jupyter notebooks that contain all the data manipulation, exploration, description and analyses you have carried out on the various datasets you have chosen to use. 


* These notebooks should be well organized and documented using Markdown cells around your code cells. 
    * __DON'T EXPECT THE CODE TO SPEAK FOR ITSELF__ - It is important to provide comments and documentation around the code you use to do your analysis. Notebooks and Markdown cells make this kind of programming easier to do and much better for sharing with others.


* Use a Markdown document called `README.md` to describe and link to all the notebooks in this analysis folder. 



#### 3. Your data story 

* This is the creative output that tells the main story that you have selected out of all of your analysis.


* It is up to you to decide how you want to present your data story - so __BE CREATIVE!__. 


* But possible options could be:
    * A data journalism article (like those we've seen from NYTimes, Guardian, LATimes, etc.)
    * A blog post (like those on medium.com)
    * A slide show presentation (jupyter notebooks can be used to make reveal.js web presentations)
    * A screencast style video (e.g. the hand drawing on a whiteboard style presentation)
    * Other kinds of video, e.g. vlog, story video etc.
    * A poster
    
    

---

### Tasks for lab session 11/11/19


#### Upload and copy data

1. Upload the datasheets you have found so far to the data folder.
2. If you already have the data uploaded in another folder you can:
    1. use the Jupyter move/duplicate function in the Jupyter file browser
    2. use the `!cp SOURCE DEST` shell command:
        * In a code cell you can run something like:
          ```
          !cp ../Assignment_3/data/my_data.csv data
          ```
          * The `!cp` means run the `cp` (i.e. copy) command
          * The `..` means move up one level of folders
          * The `/` indicates a folder 
          * So `../Assignment_3/data` means go up a level from the current folder (which is `COMM313_Final_Project`) then into the `Assignment_3` folder and then the `data` folder within that folder.
          * Then you can find the data file which might be called something like `my_data.csv`
          * The third part is the location you want to copy this file to, here it is the `data` folder in the current folder (which is `COMM313_Final_Project`),
          
          
3. Update the `README.md` file to list these files


* If you have a data file that is too large to upload talk to us and we can maybe help split it up into smaller chunks.


#### Start your analysis in an notebook

* If you already have started doing some investigative analysis elsewhere you can copy these notebooks into the `data_analysis` folder.




In [1]:
%matplotlib inline
import pandas as pd
import matplotlib.pyplot as plt
import math

olympics = pd.read_csv('data/athlete_events.csv') 

FileNotFoundError: [Errno 2] File b'data/athlete_events.csv' does not exist: b'data/athlete_events.csv'

In [16]:
olympics.shape

(271116, 15)

There are 271116 rows and 15 columns. The unit of analysis is event per athlete per game. 

In [17]:
olympics.head(10)

ID                      Name Sex   Age  Height  Weight            Team  \
0   1                 A Dijiang   M  24.0   180.0    80.0           China   
1   2                  A Lamusi   M  23.0   170.0    60.0           China   
2   3       Gunnar Nielsen Aaby   M  24.0     NaN     NaN         Denmark   
3   4      Edgar Lindenau Aabye   M  34.0     NaN     NaN  Denmark/Sweden   
4   5  Christine Jacoba Aaftink   F  21.0   185.0    82.0     Netherlands   
5   5  Christine Jacoba Aaftink   F  21.0   185.0    82.0     Netherlands   
6   5  Christine Jacoba Aaftink   F  25.0   185.0    82.0     Netherlands   
7   5  Christine Jacoba Aaftink   F  25.0   185.0    82.0     Netherlands   
8   5  Christine Jacoba Aaftink   F  27.0   185.0    82.0     Netherlands   
9   5  Christine Jacoba Aaftink   F  27.0   185.0    82.0     Netherlands   

   NOC        Games  Year  Season         City          Sport  \
0  CHN  1992 Summer  1992  Summer    Barcelona     Basketball   
1  CHN  2012 Summer  2012  Summer       London           Judo   
2  DEN  1920 Summer  1920  Summer    Antwerpen       Football   
3  DEN  1900 Summer  1900  Summer        Paris     Tug-Of-War   
4  NED  1988 Winter  1988  Winter      Calgary  Speed Skating   
5  NED  1988 Winter  1988  Winter      Calgary  Speed Skating   
6  NED  1992 Winter  1992  Winter  Albertville  Speed Skating   
7  NED  1992 Winter  1992  Winter  Albertville  Speed Skating   
8  NED  1994 Winter  1994  Winter  Lillehammer  Speed Skating   
9  NED  1994 Winter  1994  Winter  Lillehammer  Speed Skating   

                                Event Medal  
0         Basketball Men's Basketball   NaN  
1        Judo Men's Extra-Lightweight   NaN  
2             Football Men's Football   NaN  
3         Tug-Of-War Men's Tug-Of-War  Gold  
4    Speed Skating Women's 500 metres   NaN  
5  Speed Skating Women's 1,000 metres   NaN  
6    Speed Skating Women's 500 metres   NaN  
7  Speed Skating Women's 1,000 metres   NaN  
8    Speed Skating Women's 500 metres   NaN  
9  Speed Skating Women's 1,000 metres   NaN

**What was the first year recorded in this dataset, and what was the most recent?**

In [21]:
olympics.min()['Year']

1896

In [22]:
olympics.max()['Year']

2016

This data includes Olympics from 1896 through 2016. 

**How many Olympic games have there been?** 

In [18]:
olympics.nunique()['Games']

51

**How many unique Olympic athletes have there been?**

In [23]:
olympics.nunique()["ID"]

135571

There have been 135,571 athletes that have participated in the Olympics since 1896.

**What countries have participated?**

In [33]:
olympics["Team"].unique() 

AttributeError: 'numpy.ndarray' object has no attribute 'length'